### ISSUE TITLE
No registered kernels for SobolSample.

### ISSUE THREAD

**Issue by mcourteaux** [tensorflow/tensorflow#46957]
I tested this on two similar machines (with slightly different versions, apparently)

**System information**
- Have I written custom code (as opposed to using a stock example script provided in TensorFlow): no
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): Ubuntu 18.04 LTS
- TensorFlow installed from (source or binary): pip3 binary
- TensorFlow version (use command below): 2.3.0 and 2.3.1
- Python version: 3.6.8 and 3.8.0
- CUDA/cuDNN version: not relevant
- GPU model and memory: not relevant; but GTX1660Ti and RTX2080Ti

v2.3.0-54-gfcc4b966f1
v2.3.0-rc2-23-gb36436b087

**Describe the current behavior**
`tf.math.sobol_sample(4, 10)` produces following error:
```py
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/usr/local/lib/python3.8/dist-packages/tensorflow/python/util/dispatch.py", line 201, in wrapper
    return target(*args, **kwargs)
  File "/usr/local/lib/python3.8/dist-packages/tensorflow/python/ops/math_ops.py", line 4846, in sobol_sample
    return gen_math_ops.sobol_sample(dim, num_results, skip, dtype=dtype)
  File "/usr/local/lib/python3.8/dist-packages/tensorflow/python/ops/gen_math_ops.py", line 9280, in sobol_sample
    return sobol_sample_eager_fallback(
  File "/usr/local/lib/python3.8/dist-packages/tensorflow/python/ops/gen_math_ops.py", line 9312, in sobol_sample_eager_fallback
    _result = _execute.execute(b"SobolSample", 1, inputs=_inputs_flat,
  File "/usr/local/lib/python3.8/dist-packages/tensorflow/python/eager/execute.py", line 59, in quick_execute
    tensors = pywrap_tfe.TFE_Py_Execute(ctx._handle, device_name, op_name,
tensorflow.python.framework.errors_impl.NotFoundError: Could not find device for node: {{node SobolSample}} = SobolSample[dtype=DT_FLOAT]
All kernels registered for op SobolSample:
  <no registered kernels>
 [Op:SobolSample]
```

**Describe the expected behavior**
I don't know why this kernel is not registered. It seems the code is there (although I can't validate the correctness of it):
https://github.com/tensorflow/tensorflow/blob/v2.4.1/tensorflow/core/kernels/sobol_op.cc#L175

**Standalone code to reproduce the issue**
```py
import tensorflow as tf
tf.math.sobol_sample(4, 10)
```

**Other info / logs** Include any logs or source code that would be helpful to
diagnose the problem. If including tracebacks, please include the full
traceback. Large logs and files should be attached.

**Comment by Saduf2019** [comment]
@mcourteaux 
I ran the code shared and this has been fixed in tf 2.4 and nightly, please refer tot he [gist here](https://colab.research.google.com/gist/Saduf2019/9befe80921f9d512d5f18946d260cae0/untitled526.ipynb).

**Comment by mcourteaux** [comment]
Oooh great news! I dared to update and after getting rid of the old version (which got was getting selected for loading so libraries for some reason), sobol now indeed works! Thank you very much! Did you fix this because of this issue, or was this already fixed in 2.4 before I opened this issue?

**Comment by Saduf2019** [comment]
@mcourteaux 
It was fixed when i checked it on rf 2.4, kindly move this issue to closed status as resolved.

**Comment by mcourteaux** [comment]
Ok cool!
